src/App.tsx

```diff
@@ -1,24 +1,304 @@
+            findTodo.completed = !findTodo.completed;
+
+            return findTodo;
+          }
+
+          return todo;
+        });
+
+        setAllTodos(newTodos);
+      } catch {
+        setErrorMessage(ErrorMessageType.UPDATE);
+      } finally {
+        setLoadingTodoIds([]);
+      }
+    }, [allTodos],
+  );
+
+  const handleChangeStatusAllTodos = useCallback(
+    async () => {
+      const allTodosPromises = allTodos.map(async (todo) => {
+        if (hasActiveTodo) {
+          if (!todo.completed) {
+            await setLoadingTodoIds(prev => [...prev, todo.id]);
+          }
+        } else if (todo.completed) {
+          await setLoadingTodoIds(prev => [...prev, todo.id]);
+        }
+
+        return updateTodoCompleted(todo.id, hasActiveTodo);
+      });
+
+      try {
+        const updatedTodos = await Promise.all(allTodosPromises);
+
+        setAllTodos(updatedTodos);
+      } catch {
+        showError(ErrorMessageType.UPDATE);
+      } finally {
+        setLoadingTodoIds([]);
+      }
+    }, [allTodos],
+  );
+
+  const changeTitleByDoubleClick = useCallback(
+    async (todoId: number, newTitle: string) => {
+      if (!newTitle.length) {
+        await handleRemoveTodo(todoId);
+
+        return;
+      }
+
+      try {
+        await setLoadingTodoIds(prev => [...prev, todoId]);
+
+        await updateTodoTitle(todoId, newTitle);
+        const newTodos = allTodos.map((todo) => {
+          if (todoId === todo.id) {
+            const updatedTodo = { ...todo };
+
+            updatedTodo.title = newTitle;
+
+            return updatedTodo;
+          }
+
+          return todo;
+        });
+
+        setAllTodos(newTodos);
+      } catch {
+        showError(ErrorMessageType.UPDATE);
+      } finally {
+        setLoadingTodoIds([]);
+      }
+    }, [allTodos],
+  );
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            aria-label="clear"
+            type="button"
+            onClick={handleChangeStatusAllTodos}
+            className={classNames(
+              'todoapp__toggle-all',
+              {
+                active: hasActiveTodo,
+              },
+            )}
+          />
+
+          <Form
+            handleAddTodo={handleAddNewTodo}
+            todoTitle={todoTitle}
+            setTodoTitle={setTodoTitle}
+            activeInput={activeInput}
+          />
+        </header>
+
+        {(!allTodos.length && isRequest && !errorMessage)
```

Comment:
this would be enough
```suggestion
        {(!allTodos.length
```